include/framework/everest.hpp

```diff
@@ -87,26 +93,32 @@ class Everest {
     ///
     void publish_var(const std::string& impl_id, const std::string& var_name, nlohmann::json value);
 
+    ///
+    /// \brief Publishes a variable at the given \p _index, of the given \p impl_id, names \p var_name with the given \p
+    /// value
+    ///
+    void publish_var(std::size_t index, const std::string& impl_id, const std::string& var_name, nlohmann::json value);
+
     ///
     /// \brief Subscribes to a variable of another module identified by the given \p req and variable name \p
     /// var_name. The given \p callback is called when a new value becomes available
     ///
     void subscribe_var(const Requirement& req, const std::string& var_name, const JsonCallback& callback);
 
     ///
-    /// \brief Return the error manager for the given \p impl_id
+    /// \brief Return the error managers for the given \p impl_id
     ///
-    std::shared_ptr<error::ErrorManagerImpl> get_error_manager_impl(const std::string& impl_id);
+    std::vector<std::shared_ptr<error::ErrorManagerImpl>> get_error_manager_impl(const std::string& impl_id);
```

Comment:
then the signatures should be changed to `get_error_manager_impls` (also for the 3 below)